diesel/src/pg/types/date_and_time/chrono.rs

```diff
@@ -139,6 +139,48 @@ impl FromSql<Date, Pg> for NaiveDate {
     }
 }
 
+const DAYS_PER_MONTH: i32 = 30;
```

Comment:
that seems arbitrary

Reply:
Please read line 175~178. This is a ratio when the Postgres explicitly converts between months and days.

Reply:
I agree these are the most sensible values.

What I mean is that `'2024-05-25'::Timestamp + '1 month'::interval` != `DateTime::from_sql('2024-05-25'::Timestamp) + chrono::TimeDelta::from_sql('1 month'::interval)`, and `ToSql(chrono::TimeDelta::from_sql('1 month'::interval)) != '1 month'::interval`, and I'm wondering if this behavior doesn't make for enough of a trap that we should instead have users deserialize to [`PgInterval`](https://docs.diesel.rs/master/diesel/pg/data_types/struct.PgInterval.html) and just provide functions that make conversions from that to `chrono::TimeDelta` reasonably easy but more explicit on that it may lose information and behave differently.

Reply:
As far as I know there isn't any separated dimension or field for months in `chrono::Duration`. I don't think any other widely used Rust "timedelta" types support it. It is fundamental behavior difference between PostgreSQL interval and widely used (and standard in my opinion) interval data types.

PostgreSQL returns timedelta with "0 month" when it is produced by `datetime - datetime`. For example `SELECT '2024-06-26'::Timestamp - '2024-05-26'::Timestamp` will return `'31 days'::interval` instead of `'1 month'::interval`. The month field is used only when the user explicitly specifies it(would be specifying SQL literals for diesel users), so the error case you mentioned occurs on only who uses "month" dimension in their SQL query explicitly.

I am not sure if there would be many cases which uses month field explicitly, as PostgreSQL also warns these special behaviors based on political stuffs like daylight savings in their official documentation. In my opinion all users who use "month" dimension in interval should know their different behaviors.

Also this PR does not delete usage of `PgInterval`, so some sensitive users may still use it if they want. However I agree it might makes sense to document there could be some information loss.

Reply:
@McDic Could you open a PR for the documentation clarification?

Reply:
@weiznich Sure. I will create a PR. Is this urgent?